### PR TITLE
test: add e2e extended mode test

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

1. Add extended mode test.
2. Refine original extended mode test in e2e test.

related issue: #8560 #8112

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/tests/e2e_extended_mode/README.md
Comment (xxchan) [inline src/tests/e2e_extended_mode/README.md:1]: Thanks for writing this 😄

**Review by xxchan — COMMENTED** [review]
Generally lgtm

File: e2e_test/extended_mode/basic.slt [hunk 1]
```diff
@@ -1,14 +1,4 @@
...
-# There are two kinds of statement, they run different path of pgwire:
-#   1. un-query statement: SET,CREATE,INSERT,FLUSH,EXPLAIN,DROP..
-#   2. query statement: SELECT,WITH,VALUES,SHOW,DESCRIBE..
-# 
-# We also need to test different type in extended query mode:
-#   smallint,int,bigint
-#   real,double precision,numeric
-#   time,date,timestamp
-
+# Test different statements(DDL,DQL,DML) in extende mode.
```
Comment (kwannoel) [inline e2e_test/extended_mode/basic.slt:1]:
```suggestion
# Test different statements(DDL,DQL,DML) in extended mode.
```

File: e2e_test/extended_mode/basic.slt [hunk 2]
```diff
@@ -78,18 +68,3 @@ with t as (select generate_series(1,3,1)) select * from t;
...
-
-query III
-select 42::real,42::double precision,42::decimal;
-----
-42 42 42
-
-query TTT
-select '20:55:12'::time,'2022-07-12'::date,'2022-07-12 20:55:12'::timestamp;
-----
-20:55:12 2022-07-12 2022-07-12 20:55:12
```
Comment (kwannoel) [inline e2e_test/extended_mode/basic.slt:95]: Hmm why do we remove these?

File: src/tests/e2e_extended_mode/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+mod opts;
+mod test;
+
+use std::process::exit;
+
+use clap::Parser;
+use tracing::{error, info};
+
+use crate::opts::Opts;
+use crate::test::TestSuit;
```
Comment (kwannoel) [inline src/tests/e2e_extended_mode/src/main.rs:24]:
```suggestion
use crate::test::TestSuite;
```

File: src/tests/e2e_extended_mode/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+async fn main() {
+    exit(run_test().await)
+}
+
+async fn run_test() -> i32 {
+    let opts = Opts::parse();
+
+    tracing_subscriber::fmt::init();
+
+    let test_suit = TestSuit::new(opts);
```
Comment (kwannoel) [inline src/tests/e2e_extended_mode/src/main.rs:36]:
```suggestion
    let test_suit = TestSuite::new(opts);
```

File: src/tests/e2e_extended_mode/src/test.rs [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+use chrono::{DateTime, NaiveDate, NaiveDateTime, NaiveTime, Utc};
+use pg_interval::Interval;
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::Decimal;
+use tokio_postgres::NoTls;
+
+use crate::opts::Opts;
+
+pub struct TestSuit {
```
Comment (kwannoel) [inline src/tests/e2e_extended_mode/src/test.rs:23]:
```suggestion
pub struct TestSuite {
```

File: src/tests/e2e_extended_mode/src/test.rs [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    config: String,
+}
+
+impl TestSuit {
```
Comment (kwannoel) [inline src/tests/e2e_extended_mode/src/test.rs:27]:
```suggestion
impl TestSuite {
```

**Review by kwannoel — APPROVED** [review]
LGTM! Thanks for this

File: e2e_test/extended_mode/basic.slt (hunk 2 above)
Comment (ZENOTME) [reply e2e_test/extended_mode/basic.slt:95]: We have cover it with extended_mode/type.slt.🤔

**Pull request merged**